ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/avdinterfacedescriptions.py

```diff
@@ -42,8 +42,10 @@ def underlay_ethernet_interface(self, data: InterfaceDescriptionData) -> str:
             - mpls_lsr
             - overlay_routing_protocol
             - type
+            - vrf
         """
-        return self.underlay_ethernet_interfaces(link_type=data.link_type, link_peer=data.peer, link_peer_interface=data.peer_interface)
+        desc = self.underlay_ethernet_interfaces(link_type=data.link_type, link_peer=data.peer, link_peer_interface=data.peer_interface)
+        return f"{desc} vrf {data.vrf}" if data.vrf is not None else desc
```

Comment:
This means someone else cannot override this with Jinja2 templates. It is probably the best we can do for now, but it _is_ a caveat.

Reply:
I had a version with `vrf` as a keyword argument to `self.underlay_ethernet_interface` before with a default of None, I can change it to this if you prefer?

Reply:
No because that would also fail for someone using a custom subclass.